pandas/tests/extension/test_string.py

```diff
@@ -237,9 +241,11 @@ def test_arith_series_with_array(
             using_infer_string
             and all_arithmetic_operators == "__radd__"
             and (
-                (dtype.na_value is pd.NA) or (dtype.storage == "python" and HAS_PYARROW)
+                dtype.na_value is pd.NA
+                and not (not HAS_PYARROW and dtype.storage == "python")
             )
```

Comment:
Nit: This would be a bit cleaner as 

```
using_infer_string
and all_arithmetic_operators == "__radd__"
and dtype.na_value is pd.NA
and (HAS_PYARROW or dtype.storage == "pyarrow")
```